### DISCUSSION TITLE
Mocking modules in tests using RuleTester

### DISCUSSION THREAD

**Discussion by jsaelhof** [eslint/eslint#17401] [Q&A] [answered]
I'm working on building an ESLint plugin that tests for the presence of other files in the same directory (ex: File is missing a corresponding test file). The rule works fine and uses Node's `fs` module to read the directory and look for files. The issue I'm having is that while testing using the RuleTester, I can provide a filename for the test case but I have no way of providing the names of the other files. My thought was to mock `readdirSync` (used within my rule's code) to provide them. I'm using vitest and the mock registers but RuleTester uses the real implementation. Is there any way to mock modules for the RuleTester?

**Comment by nzakas** [comment, marked as answer]
No there isn't. ESLint isn't intended to be used to lint the file system so we don't have any way to swap out the default `fs`.
